notebook_data_redirector/sync.py

```diff
@@ -11,13 +11,21 @@ def lambda_handler(event, context):
     ddb_table = common.get_ddb_table()
     box_client, _ = common.get_box_client()
     root_folder = box_client.folder(common.BOX_FOLDER_ID)
+    root_shared = common.is_box_file_public(root_folder)
 
     LOGGER.info("Checking files in Box")
     shared_file_ids = set()
     shared_filepaths = set()
     count = 0
-    for file in common.iterate_files(root_folder):
+    for file, shared in common.iterate_files(root_folder, shared=root_shared):
         count += 1
+        if (not common.is_box_file_public(file)) and shared:
+            # this includes an api call
+            if file.type == "file":
+                file = common.create_shared_link(box_client, file, access=u"open", allow_download=True)
+        if (common.is_box_file_public(file)) and (not shared):
```

Comment:
((operator precedence) makes (the parens) unnecessary, but I imagine this was done more for readability?)

Reply:
This could be an `elif`, since it shouldn't be possible for both conditions to be true

Reply:
I'm in the habit of being very explicit with logical groupings, for readability, and also probably because I don't always know what the operator precedence actually is...